src/sqlalchemy_cratedb/dialect.py

```diff
@@ -206,6 +206,15 @@ def __init__(self, **kwargs):
         # start with _. Adding it here causes sqlalchemy to quote such columns.
         self.identifier_preparer.illegal_initial_characters.add("_")
 
+    def get_isolation_level_values(self, dbapi_conn):
+        return ()
+
+    def set_isolation_level(self, dbapi_connection, level):
+        pass
+
+    def get_isolation_level(self, dbapi_connection):
+        return "NONE"
```

Comment:
Any recommendations?

Reply:
Consulting Wikipedia and the SQLAlchemy documentation, `READ UNCOMMITTED` might be the right choice for CrateDB?

- https://en.wikipedia.org/wiki/Isolation_%28database_systems%29#Read_uncommitted
- https://github.com/sqlalchemy/sqlalchemy/blob/rel_2_0_40/lib/sqlalchemy/engine/interfaces.py#L2487-L2529

Reply:
Typical PostgreSQL drivers in Java (JDBC) and Python offer the same quartet, and Apache Spark additionally seems to provide `NONE`.

- [JDBC](https://www.unrepo.com/jdb/transaction-isolation-levels-in-jdbc)
- [psycopg3](https://github.com/psycopg/psycopg/blob/3.2.6/psycopg/psycopg/_enums.py#L55-L69)
- [asyncpg](https://github.com/MagicStack/asyncpg/blob/v0.30.0/asyncpg/transaction.py#L22-L33)
- [Apache Spark](https://spark.apache.org/docs/latest/sql-data-sources-jdbc.html)

So, I guess using `READ UNCOMMITTED` as the default and single valid choice for CrateDB is the right way to resolve the situation?

Reply:
It looks like specifically for SQLAlchemy, `AUTOCOMMIT` seems to be more appropriate.

> SQLAlchemy treats the concept of “autocommit” like any other isolation level; in that it is an isolation level that loses not only “read committed” but also loses atomicity.

-- https://docs.sqlalchemy.org/en/20/core/connections.html#dbapi-autocommit